README.md

```diff
@@ -1 +1,167 @@
+
+```js
+var processResponse1 = require('./response1.js')
+callOne({...}, processResponse1)
+```
+
+This is the `response1.js` file:
+
+```js
+var processResponse2 = require('./response2.js')
+module.exports = function processResponse1(error, data1) {
+  callTwo(data1, processResponse2)
+}
+```
+
+This is the `response2.js` file:
+
+```js
+var processResponse3 = require('./response3.js')
+module.exports = function processResponse2(error, data2) {
+  callThere(data2, processResponse3)
+}
+```
+
+This is the `response3.js` file:
+
+```js
+module.exports = function processResponse3(error, data3) {
+  ...
+}
+```
+
+
+So now your 5 co-workers will love you even more because each piece of logic will be in its own file (you can easily separate files into different folder by providing paths to the `require`). Another added benefit of modular approach is that you and your team can write unit tests for each module. Having unit tests greatly increases robustness of the system and the velocity of development, because developers can make changes without being afraid of unintentionally breaking existing functionality (if they break the tests will catch that!).
+
+## Callback Arguments Convention
+
+Typically callbacks follow `error, result` convention where `error` is an error object (null is there's no errors) and `result` which is what you would expect from a function. Because this is just a convention, libraries and frameworks don't have to follow it. That's why you must read the usage for each library before attempting to use it's methods.
+
+The names you use in the callbacks for arguments such as `error` or `result` don't really matter. The best practice is to make them meaningful to avoid confusion (`error` is better to read than `err` or `e` which might mean an event to other developers working on this code). What matters in the callback declarations is **order** of arguments.
+
+Lastly, the callback function is typically the last argument. It's another convention, so it's not enforced by the Node platform in any way. It's just the best practice to follow. Again, you will often see different names for callbacks such as `cb`, `done`, `next`, `end`, and our favorite `callback` (because it's easier to read and understand).
+
+## Resources
+
+1. [Callback Hell, where bad code goes to rest](http://callbackhell.com)
+2. [Concurrency vs. Parallelism](http://stackoverflow.com/questions/1050222/concurrency-vs-parallelism-what-is-the-difference#1050257)
+3. [Avoiding Callback Hell in Node.js](http://stackabuse.com/avoiding-callback-hell-in-node-js)
+4. [Managing Node.js Callback Hell with Promises, Generators and Other Approaches](https://strongloop.com/strongblog/node-js-callback-hell-promises-generators)
```

Comment:
I actually really like that we're giving this to students as a resource early. Maybe if we can cut that section above we can give a forward-looking paragraph or two about promises and generators?